sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_client_async.py

```diff
@@ -828,8 +828,11 @@ def get_buffered_event_count(self, partition_id: str) -> Optional[int]:
         """
         The number of events that are buffered and waiting to be published for a given partition.
          Returns None in non-buffered mode. **NOTE: This method should only be used for debugging
-         purposes. If an invalid `partition_id` is passed in, then a value of 0 will be returned
-         which may be misleading.**
+         purposes. Returns the approximate number of events in the buffer for a partition ID,
+         because it is reporting the state of a background thread.  For a partition ID that has no
```

Comment:
Background coroutine? Or did we use threads for async as well?

Reply:
I think it should be background coroutine @kashifkhan?